src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,87 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
 
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private static final int RESIZE_VALUE = 2;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
+
     @Override
     public void put(K key, V value) {
-
+        if (size == table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> nextNode = table[index];
+            Node<K, V> prevNode = null;
+            while (nextNode != null) {
+                if (Objects.equals(key, nextNode.key)) {
+                    nextNode.value = value;
+                    return;
+                }
+                prevNode = nextNode;
+                nextNode = nextNode.next;
+            }
+            prevNode.next = newNode;
+        }
+        size++;
```

Comment:
if key already exists, size will not be changed
```suggestion
            table[index] = newNode;
            size++;
        } else {
            Node<K, V> nextNode = table[index];
            Node<K, V> prevNode = null;
            while (nextNode != null) {
                if (Objects.equals(key, nextNode.key)) {
                    nextNode.value = value;
                    return;
                }
                prevNode = nextNode;
                nextNode = nextNode.next;
            }
            prevNode.next = newNode;
        }
```